keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -148,6 +146,21 @@ class ArtifactListener(
   private fun getLatestStoredVersion(artifact: DeliveryArtifact): String? =
     repository.artifactVersions(artifact).sortedWith(artifact.versioningStrategy.comparator).firstOrNull()
 
+  /**
+   * Returns a copy of the [PublishedArtifact] with the git and build metadata populated, if available.
+   */
+  private fun ArtifactSupplier<*,*>.addMetadata(artifact: PublishedArtifact): PublishedArtifact {
+    val artifactMetadata = runBlocking {
+      try {
+        getArtifactMetadata(artifact)
```

Comment:
Let's add retries around this, please

Reply:
The retries are already there, further down into the call chain, in `ArtifactMetadataService`.

Reply:
GREAT!